src/libolm_migration.rs

```diff
@@ -0,0 +1,446 @@
+) -> anyhow::Result<()> {
+    let account = store
+        .load_account()
+        .await?
+        .context("Base data must be imported before calling `migrateOlmSessions`")?;
+
+    let user_id = account.user_id();
+    let device_id = account.device_id();
+    let identity_keys = &account.identity_keys;
+
+    let sessions = pickled_sessions
+        .into_iter()
+        .map(|pickled_session| {
+            Session::from_pickle(
+                user_id.to_owned(),
+                device_id.to_owned(),
+                identity_keys.clone(),
+                pickled_session,
+            )
+        })
+        .collect();
+
+    store.save_changes(Changes { sessions, ..Default::default() }).await?;
+    Ok(())
+}
+
+/// A pickled version of an `InboundGroupSession`.
+///
+/// Holds all the information that needs to be stored in a database to restore
+/// an InboundGroupSession.
+#[wasm_bindgen(getter_with_clone)]
+#[derive(Debug, Default)]
+pub struct PickledInboundGroupSession {
+    /// The pickle string holding the Megolm Session, as returned by
+    /// `olm_pickle_inbound_group_session` in libolm.
+    pub pickle: String,
+
+    /// The public curve25519 key of the account that sent us the session.
+    #[wasm_bindgen(js_name = "senderKey")]
+    pub sender_key: String,
+
+    /// The public ed25519 key of the account that sent us the session.
+    #[wasm_bindgen(js_name = "senderSigningKey")]
+    pub sender_signing_key: String,
+
+    /// The id of the room that the session is used in.
+    ///
+    /// Nullable so that a `PickledInboundGroupSession` can be constructed
+    /// incrementally. Must be populated!
+    #[wasm_bindgen(js_name = "roomId")]
+    pub room_id: Option<RoomId>,
+
+    /// Flag remembering if the session was directly sent to us by the sender
+    /// or if it was imported.
+    pub imported: bool,
+
+    /// Flag remembering if the session has been backed up.
+    #[wasm_bindgen(js_name = "backedUp")]
+    pub backed_up: bool,
+}
+
+#[wasm_bindgen]
+impl PickledInboundGroupSession {
+    /// Construct a new `PickledInboundGroupSession`, with default values.
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[wasm_bindgen]
+impl Migration {
+    /// Migrate Megolm sessions of a libolm-based setup to a vodozemac-based setup
+    /// stored in an indexedDB crypto store.
+    ///
+    /// Before this method can be used, {@link #migrateBaseData} must be used to
+    /// import the base data into the crypto store.
+    ///
+    /// # Arguments
+    ///
+    /// * `sessions` - An `Array` of {@link PickledInboundGroupSession}s to import.
+    /// * `pickle_key` - The libolm pickle key that was used to pickle the megolm
+    ///   session objects.
+    /// * `store_handle` - A connection to the CryptoStore which will be used to
+    ///   store the vodozemac data.
+    #[wasm_bindgen(js_name = "migrateMegolmSessions")]
+    pub async fn migrate_megolm_sessions(
```

Comment:
This could be potentially a lot of sessions. Should we inform the caller to do that in chunks? Or have some progress reporting?

Reply:
The intention is that the application calls this with small enough batches that there is no need for an intermediate progress update.

Reply:
Maybe document that?

Reply:
808a5736